src/tui/editor.zig

```diff
@@ -2113,9 +2113,9 @@ pub const Editor = struct {
         try self.move_begin(ctx);
         const b = try self.buf_for_update();
         const root = try self.delete_to(move_cursor_end, b.root, b.allocator);
+        try self.delete_forward(ctx);
```

Comment:
This will still not do what you expect it to because `delete_forward` does it's own buffer update. You want to copy 2124 here and adjust it to update the `root` variable.